collector/mongod/oplog_status.go

```diff
@@ -1,102 +1,116 @@
 package collector_mongod
 
 import (
-    "time"
-    "github.com/golang/glog"
-    "github.com/prometheus/client_golang/prometheus"
-    "gopkg.in/mgo.v2"
-    "gopkg.in/mgo.v2/bson"
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
 )
 
 var (
-    oplogStatusLengthSec = prometheus.NewGauge(prometheus.GaugeOpts{
-            Namespace: Namespace,
-            Subsystem: "oplog",
-            Name:      "length_sec",
-            Help:      "Length of oplog in seconds from head to tail",
-    })
-    oplogStatusLengthSecNow = prometheus.NewGauge(prometheus.GaugeOpts{
-            Namespace: Namespace,
-            Subsystem: "oplog",
-            Name:      "length_sec_now",
-            Help:      "Length of oplog in seconds from now to tail",
-    })
-    oplogStatusSizeMB = prometheus.NewGauge(prometheus.GaugeOpts{
-            Namespace: Namespace,
-            Subsystem: "oplog",
-            Name:      "size_mb",
-            Help:      "Size of oplog in megabytes",
-    })
+	oplogStatusCount = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace:	Namespace,
+		Subsystem:	"replset_oplog",
+		Name:		"items_total",
+		Help:		"The total number of changes in the oplog",
+	})
+	oplogStatusHeadTimestamp = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace:	Namespace,
+		Subsystem:	"replset_oplog",
+		Name:		"head_timestamp",
+		Help:		"The timestamp of the newest change in the oplog",
+	})
+	oplogStatusTailTimestamp = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace:	Namespace,
+		Subsystem:	"replset_oplog",
+		Name:		"tail_timestamp",
+		Help:		"The timestamp of the oldest change in the oplog",
+	})
+	oplogStatusSizeBytes = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace:	Namespace,
+		Subsystem:	"replset_oplog",
+		Name:		"size_bytes",
+		Help:		"Size of oplog in bytes",
+	}, []string{"type"})
 )
 
-func GetCollectionSizeMB(db string, collection string, session *mgo.Session) (float64) {
```

Comment:
The only lowlevel datatype in prometheus is float64, everything has to be one even if it's a 0/1. It won't accept regular float/int or int64.